### PR TITLE
New fixes for https://github.com/Exa-Networks/exabgp/pull/627

### PR DESCRIPTION
**Pull request by leleobhz**
* Automatic version set
* Requires line fixes
* Changelog automatically generated

---
This change is [Reviewable](https://reviewable.io/reviews/exa-networks/exabgp/629)

### CODE REVIEW

**Comment by leleobhz** [comment]
Resulting RPM:

* Copy exabgp-3.4.10_954_gd13aa89-11494527190.fc25.noarch.rpm
* Copy python-exabgp-3.4.10_954_gd13aa89-11494527190.fc25.noarch.rpm

**Comment by thomas-mangin** [comment]
on master you can not reliably use git to extract the tags and get the current (and next version).
So I added to the setup.py script the "current" and "next" command.

```
$ python setup.py current
3.4.18
$ python setup.py next
4.0.0
```

The next is not really useful here tho. Can you please update the code to use the feature please ?

**Comment by thomas-mangin** [comment]
To explain the 3.4 releases are done from the 3.4 branch and not master.

**Comment by leleobhz** [comment]
@thomas-mangin Sure! I'll do it.

**Comment by leleobhz** [comment]
> To explain the 3.4 releases are done from the 3.4 branch and not master.

About these specs, I've explained what I plan in https://github.com/Exa-Networks/exabgp/issues/626#issuecomment-301129337 - targeting 2 specs - One to be sent to distributions, other to help bleeding-edge users to do not mess their systems. What do you think about it?

**Comment by thomas-mangin** [comment]
Just for clarification the code for 3.4 is not on master so you can not build 3.4.18 using master, so I will need to backport setup.py to 3.4 if you want to use this branch and the this script to build the RPM

**Comment by leleobhz** [comment]
@thomas-mangin I've changed the PR to attend what you asked me.

A question: Do you recommend/can create a "default" exabgp.conf that only starts the daemon and does notting? I'm using ./examples/api-api.conf for now to test the instalation.

File: redhat/python-exabgp.spec.git [hunk 1]
```diff
@@ -1,20 +1,29 @@
 %{!?__python2:        %global __python2 /usr/bin/python2}
 %{!?python2_sitelib:  %global python2_sitelib %(%{__python2} -c "from distutils.sysconfig import get_python_lib; print(get_python_lib())")}
+%define version %(echo "$(python setup.py current).git$(git rev-parse --short HEAD)")
```
Comment (thomas-mangin) [inline redhat/python-exabgp.spec.git]: I am confused by this line ...

Comment (leleobhz) [inline redhat/python-exabgp.spec.git]: Executing in a shell:

captura de tela de 2017-05-12 15-45-08

The puporse of this line is populate a version numer with you corrected syntax + git<COMMIT> recommendation from Fedora - as in https://fedoraproject.org/wiki/Packaging:Versioning#Snapshots

**Comment by thomas-mangin** [comment]
ok - ported the changes from master to 3.4 can you please use the right tag'ed releases ? 
```
$ git rev-parse --short $(python setup.py current)
521038d5
```

**Comment by thomas-mangin** [comment]
That would be for the latest stable release using the 3.4 branch. You may also add another spec file to build HEAD (for both 3.4 and master) .. that would make sense.

File: redhat/python-exabgp.spec.git (hunk 1 above)
Comment (thomas-mangin) [inline redhat/python-exabgp.spec.git]: `%define version %(echo "$(python setup.py current).git$(git rev-parse --short $(python setup.py current))")`

Comment (thomas-mangin) [inline redhat/python-exabgp.spec.git]: That would work in 3.4 but would not be good for master tho ... master does not have any recent 3.4.xx tag and code. The code forked at 3.4.10 and all the fix were forward ported.

Comment (thomas-mangin) [inline redhat/python-exabgp.spec.git]: The problem is the the tag is not available on the master branch for 3.4.xxx

**Comment by leleobhz** [comment]
Hello @thomas-mangin !

As I said in another comment, I'll find ways to do this spec more portable. :)

**Pull request merged**